frontend/src/components/mods/PresetsManagement.tsx

```diff
@@ -177,6 +182,29 @@ export default function PresetsManagement() {
         }
     }
 
+    const handlePresedDialogOpen = (presetId: string) => {
+        setSelectedPresetId(presetId);
+        setRenamePresetDialogOpen(true);
+    }
+
+    const handlePresedDialogClose = () => {
+        setRenamePresetDialogOpen(false);
+        setSelectedPresetId(null);
+    }
+
+    const handleRenameNewPreset = async (presetName: string, presetId: string) => {
+        setRenamePresetDialogOpen(false);
+        setSelectedPresetId(null);
+
+        const request = {
+            name: presetName,
+        }
+
+        await renameModPreset(presetId, request);
+        toast.success(`Preset '${presetName}' successfully renamed`);
+        window.location.reload();
```

Comment:
Using `window.location.reload()` is not ideal to re-render the changes. Instead, you should modify the state of the component, in this case the `presets` state.

Something like this should work:
```typescript
const newPresets = [...presets]; // create a copy of the original state
const renamedPreset = newPresets.find(preset => preset.id === presetId); // find the renamed preset by ID
renamedPreset.name = presetName; // edit the preset name
setPresets(newPresets); // set the new state with renamed preset
```

This should automatically re-render the component with the new state.